### PR TITLE
olligranlund keymap updates for DZ60 and Keebio/Iris

### PR DESCRIPTION
**Pull request by OlliGranlund**
I've reworked my keymaps for both Iris and DZ60. 

### DZ60
I created a new keymap for my second DZ60 board which has a different layout than my first one. Added VIA support.

### Keebio/Iris
I created a new keymap for the rebuild of my Iris board. This layout differs to some extent from the original and tries to be more inline with any normal 60% ISO keyboard. Added VIA support.

## Description

The changes done are only keymap changes to keymaps which I've uploaded earlier.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
In its current form, this PR cannot be accepted due to the submodule changes in `lib/vusb`.

**Comment by OlliGranlund** [comment]
I guess I had an too old version of qmk fork, merged now from upstream latest master

**Comment by zvecr** [comment]
image

Still no good.

**Comment by OlliGranlund** [comment]
Updated all the submodules with this command:

`git submodule foreach git pull origin master`

**Comment by zvecr** [comment]
Now you have `lib/googletest` and `lib/lufa` changes

**Comment by OlliGranlund** [comment]
Apparently updating the submodules turned out to be trickier than I expected... Where do we have stated which versions of the submodules should be used? I redownloaded QMK and replaced the lib-folder with the latest generated by qmk, but still running into the same issue...

Any hints on how I should proceed?

**Comment by OlliGranlund** [comment]
Got some help from QMK discord to cleanup the repo, no changes to git submodules anymore :)

File: keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c [hunk 1]
```diff
@@ -0,0 +1,126 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+extern keymap_config_t keymap_config;
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c:3]: This is not needed.
```suggestion
```

File: keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+#define _QWERTY 0
+#define _LOWER 1
+#define _RAISE 2
+#define _ADJUST 16
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c:8]: Should use an enum for this:
```suggestion
enum layers (
    _QWERTY,
    _LOWER,
    _RAISE
    _ADJUST,
};
```

File: keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+	//└────────┴────────┴────────┴───┬────┴───┬────┴───┬────┴───┬────┘        └───┬────┴───┬────┴───┬────┴───┬────┴────────┴────────┴────────┘
+									  _______, _______, _______,                   _______, _______, _______
+								  // └────────┴────────┴────────┘                 └────────┴────────┴────────┘
+	)
+};
+
+void persistent_default_layer_set(uint16_t default_layer) {
+	eeconfig_update_default_layer(default_layer);
+	default_layer_set(default_layer);
+}
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c:85]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keebio/iris/keymaps/olligranlund_nordic/config.h [hunk 5]
```diff
@@ -19,18 +19,18 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
 
-#define USE_SERIAL
-//#define USE_I2C
+//#define USE_SERIAL
+#define USE_I2C
 
 /* Select hand configuration */
 
-// #define MASTER_LEFT
+#define MASTER_LEFT
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/olligranlund_nordic/config.h:27]: Technically, this isn't required, as it's the default behavior.

**Review by drashna — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]

File: keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+ * You should have received a copy of the GNU General Public License 
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+ */ 
+
+#include QMK_KEYBOARD_H
+
+enum layers (
+    _QWERTY,
+    _LOWER,
+    _RAISE
```
Comment (noroadsleft) [inline keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c:22]: Fixing two typos; this keymap doesn't compile without these fixes.

```suggestion
enum layers {
    _QWERTY,
    _LOWER,
    _RAISE,
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Actually, you need this:

File: keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+};
+
+enum custom_keycodes {
+	QWERTY = SAFE_RANGE,
+	LOWER,
+	RAISE,
+	ADJUST,
+};
+
+#define KC_ KC_TRNS
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c:33]: This should be removed, as it doesn't appear to be used. 
```suggestion
```

File: keyboards/handwired/neito/config.h [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x2602
```
Comment (drashna) [inline keyboards/handwired/neito/config.h:19]: This vendor ID is already in use. 
https://www.the-sz.com/products/usbid/index.php?v=0x2602&p=&n=

```
❯ echo -n "Oliver Granlund" | shasum | cut -c1-4 | tr '[:lower:]' '[:upper:]'   
B9F7
```

```suggestion
#define VENDOR_ID       0xB9F7
```
Or if you want to use ascii
```suggestion
#define VENDOR_ID       0x4F47 // "OG"
```

File: keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+	switch (keycode) {
+		case QWERTY:
+			if (record->event.pressed) {
+				set_single_persistent_default_layer(_QWERTY);
+			}
+			return false;
+			break;
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/olligranlund_nordic_v2/keymap.c:105]: This isn't needed, since you only have the one default layer. 
```suggetion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**